### PR TITLE
Improve chapter 3

### PR DESCRIPTION
**Pull request by svenvanderburg**

### CODE REVIEW

File: _episodes/03-computing-pi.md [hunk 1]
```diff
@@ -5,21 +5,20 @@ exercises: 30
 questions:
 - "What is the GIL?"
 - "How do I parallelize an elementary program?"
-- "What is vector-based parallelism?"
-- "What is task-based parallelism?"
+- "What is data parallelism?"
```
Comment (svenvanderburg) [inline _episodes/03-computing-pi.md:8]: I can find more references to data-parallelism than vector-parallelism, so I changed it. But would be nice to get an 2nd opinion on this.

File: _episodes/03-computing-pi.md [hunk 2]
```diff
@@ -1,25 +1,24 @@
...
 - "Recognize the primitive components of the queue/worker based model of execution."
 keypoints:
 - "Vectorized algorithms are both a blessing and a curse."
 - "If we want the most efficient parallelism on a single machine, we need to unlock the GIL."
 - "Numba helps you both speeding up and lifting code from the GIL."
 ---
-FIXME: Vector-based parallelization is explained well, but task-based is
-actually missing from this chapter.
 
 FIXME: Maybe it makes sense to explain the difference between multiprocessing and
```
Comment (svenvanderburg) [inline _episodes/03-computing-pi.md:23]: I wasn't really sure where to put this without destroying the flow in the epipsode, so leaving this hanging for now.

File: _episodes/02-benchmarking.md [hunk 3]
```diff
@@ -23,12 +23,15 @@ We will get into creating parallel programs in Python later. First let's see a s
 your system monitor, and run the following:
 
 ~~~python
+# A simple summation making use of numpy's vectorization:
```
Comment (jhidding) [inline _episodes/02-benchmarking.md:26]: Try to avoid diminuative words like "simple"

File: _episodes/03-computing-pi.md [hunk 4]
```diff
@@ -200,6 +222,7 @@ We now build a queue/worker model. This is the basis of multi-threading applicat
 this point creating a parallel program is quite involved. After we've done this, we'll see ways to
 do the same in Dask without mucking about with threads directly.
 
+FIXME: Define the `sum_primes` function
```
Comment (jhidding) [inline _episodes/03-computing-pi.md:225]: good catch! In this case it should be any of the `calc_pi` variants. I copy pasted this code from a different tutorial ;)

File: _episodes/02-benchmarking.md (hunk 3 above)
Comment (svenvanderburg) [reply _episodes/02-benchmarking.md:26]: Thanks for the feedback, I'll try to remember it! (and change it here of course)

File: _episodes/03-computing-pi.md (hunk 4 above)
Comment (svenvanderburg) [reply _episodes/03-computing-pi.md]: Ok, I am now using the `calc_pi` function. Kind of curious what would happen if you used `calc_pi_dask` (I guess that threading would not actually make it faster).

**Pull request merged**